src/message-post-purchases/services/message-post-purchase.service.ts

```diff
@@ -0,0 +1,90 @@
+import { Injectable } from '@nestjs/common';
+import { MessagePostPurchase } from '../entities/message-post-purchase.entity';
+import { Repository } from 'typeorm';
+import { InjectRepository } from '@nestjs/typeorm';
+import {
+    paginate,
+    Pagination,
+    IPaginationOptions,
+} from 'nestjs-typeorm-paginate';
+
+
+
+@Injectable()
+export class MessagePostPurchaseService {
+    constructor(@InjectRepository(MessagePostPurchase) private messageRepo: Repository<MessagePostPurchase>) {
+
+    }
+
+    async paginate({page,limit}: IPaginationOptions, cartId: number): Promise<Pagination<MessagePostPurchase>> {
+        let countMessage = await this.messageRepo.query(`
+            SELECT COUNT(*) FROM messages_post_purchase
+        `);
+
+        let messages = await this.messageRepo.query(`
+        SELECT * FROM messages_post_purchase
+        WHERE messages_post_purchase.cart_id = ${cartId}
+        `);
+
+        const totalMessages = Number(countMessage[0].count);
+        const totalPages = Math.ceil(totalMessages / Number(limit));
+        const previousPagePath = (Number(page) > 1) ? `/purchase/${cartId}/messages?page=${Number(page) - 1}` : ""
+    
+
+        messages = messages.map(({
+            id,
+            cart_id,
+            created_at,
+            sent_at,
+            received_at,
+            read_at,
+            sender_id,
+            message
+        }) => {
+          return({
+            "id": id,
+            "item": {
+              "cart_id": cart_id,
+              "created_at": created_at,
+              "sent_at": sent_at,
+              "received_at": received_at,
+              "read_at": read_at,
+              "sender_id": sender_id,
+              "message": message
+            }
+          })
+        });
+
```

Comment:
Mucho codigo repetido en la funcion anterior.
Si los datos se copian iguales no tiene sentido repetir uno por uno.

```suggestion
       messages = messages.map((item) => {
          return({
            id,
            item
          })
        });
```